src/Utilities/set_eval.jl

```diff
@@ -0,0 +1,175 @@
+
+function distance_to_set(v::Real, s)
+    d = _distance(v, s)
+    return d < 0 : zero(d) : d
+end
+
+function distance_to_set(v::AbstractVector{T}, s) where {T <: Real}
+    length(v) != MOI.dimension(s) && throw(DimensionMismatch("Mismatch between v and s"))
+    d = _distance(v, s)
+    return [di < 0 : zero(T) : di for di in d]
+end
+
+_distance(v, s::MOI.LessThan) = v - s.upper
+_distance(v, s::MOI.GreaterThan) = s.lower - v
+_distance(v, s::MOI.EqualTo) = abs(v - s.value)
+
+_distance(v, s::MOI.Interval) = max(s.lower - v, v - s.upper)
+
+_distance(v::AbstractVector{<:Real}, ::MOI.Reals) = false * v
+
+_distance(v, ::MOI.Zeros) = abs.(v)
+
+_distance(v::AbstractVector{<:Real}, ::MOI.Nonnegatives) = -v
+_distance(v::AbstractVector{<:Real}, ::MOI.Nonpositives) = v
+
+# Norm cones
+
+function _distance(v::AbstractVector{<:Real}, ::MOI.NormInfinityCone)
+    t = first(v)
+    xs = v[2:end]
+    return [t - abs(x) for x in xs]
+end
+
+function _distance(v::AbstractVector{<:Real}, ::MOI.NormOneCone)
+    t = v[1]
+    xs = v[2:end]
+    return t - sum(abs, xs)
+end
+
+function _distance(v::AbstractVector{<:Real}, ::MOI.SecondOrderCone)
+    t = v[1]
+    xs = v[2:end]
+    return max(-t, dot(xs, xs) - t^2) # avoids sqrt
```

Comment:
This isn't correct. `max(0.0, norm(v[2:end], 2) - v[1])`

Reply:
`norm(v[2:end], 2) - v[1]` also works as negative values are set to zero afterwards